packages/contract/src/contract-call-script/src/main.sw

```diff
@@ -21,14 +19,33 @@ fn get_script_data_offset() -> u64 {
     is + script_length
 }
 
-fn main() {
-    let script_data_offset = get_script_data_offset();
-    let call_data_offset = script_data_offset + CONTRACT_ID_LEN + WORD_SIZE;
-    let amount_offset = script_data_offset + CONTRACT_ID_LEN;
-    let asset_id_offset = script_data_offset;
+fn get_script_data<T>() -> T {
+    let script_data_ptr = get_script_data_offset();
+    let script_data = asm(r1: script_data_ptr) {
+        r1: T
+    };
+    script_data
+}
+
+struct CallData {
+    contract_id: ContractId,
+    fn_selector: [u8; 8],
+    fn_arg_or_ptr: u64,
+}
 
-    asm(r1: call_data_offset, r2: amount_offset, r3: asset_id_offset) {
-        lw r2 r2 i0;
+fn call_contract(call_data: CallData, amount: u64, asset_id: b256) {
+    asm(r1: call_data, r2: amount, r3: asset_id) {
         call r1 r2 r3 cgas;
```

Comment:
@adlerjohn thanks for pointing this out. I miss informed here; Without `lw` the size was fixed to `u16`. And when passing a value up to `u64`, the value was parsed incorrectly.

This was fixed here:  https://github.com/FuelLabs/fuels-rs/pull/175

Reply:
Knowing that all values from `u8` to `u64` are encoded into `8 bytes`. I can see that the `lw` is not required, and maybe this could be an issue in another place where we `decode` the `amount`? Makes sense?

Reply:
@adlerjohn @AlicanC as we have tests to return the amount from context and this test is now working with values like `1_000_000` without the `lw` we can safely remove it.

Reply:
Also open a issue to remove it on `fuels-rs`: https://github.com/FuelLabs/fuels-rs/issues/285

Reply:
Not sure I follow. `lw` is required for integers that don't fit into 12 bits, no?